reporting/performance/test_kn.cpp

```diff
@@ -48,7 +48,9 @@ static const std::array<std::array<T, 3>, 9> kn_data = { {
    { { SC_(-10.0), SC_(1.0), SC_(1.80713289901029454691597861302340015908245782948536080022119e8) } },
    { { SC_(100.0), SC_(5.0), SC_(7.03986019306167654653386616796116726248616158936088056952477e115) } },
    { { SC_(100.0), SC_(80.0), SC_(8.39287107246490782848985384895907681748152272748337807033319e-12) } },
-   { { SC_(-1000.0), SC_(700.0), SC_(6.51561979144735818903553852606383312984409361984128221539405e-31) } },
```

Comment:
Given that the answer is within a sensible range this should work and give the correct answer for all T.  Do you know where in the code this is failing?

Reply:
> where in the code this is failing?

Yes John (@jzmaddock) that was a tough one. Internally the calculation must compute $e^{-700}$ which underflows the type.

Reply:
Sure, but it works for type double which also can't compute e^-700.  Ah wait, are these calling the 113-bit precision rational approximations?

Reply:
> Ah wait, are these calling the 113-bit precision rational approximations

Yes John. This is a conversation I was kind of waiting to have. My exponent range is EVEN less than that of `double`. But I end up in the multiprecision section of the special functions.

This worked surprisingly well and there seem to be very few errors. There are, however, a few we will need to hash out.

Reply:
This somewhat novel type has $106$ digits, yet is limited to `min_exponent10` of $-291$ and `max_exponent10` of $+308$